### PR TITLE
driver for HP4142B derived from B1500A

### PR DESCRIPTION
**Pull request by matthias6**
Hi,

i am very happy, that i came across the pymeasure project.
For the old HP4142A Modular DC Source i took the Agilent B1500A from M. Jung and adapted it to HP4142B.
It has nearly the same syntax.

Looking forward to your feedback.

Matthias

### CODE REVIEW

**Comment by bilderbuchi** [comment]
Thank you, that commit fd6f026 is quite useful to see the difference. From a quick review it definitely looks like we should derive one from the other (or both from one superclass). Many differences like parameter sets are or can be put in class attributes, some can be accomodated with a bit of restructuring. We should definitely _not_ just copy-paste, here. 
Help wanted if somebody (ideally having one or both devices available) can take a stab at this!

File: pymeasure/instruments/hp/hp4142B.py [hunk 1]
```diff
@@ -0,0 +1,2012 @@
...
+        or file headers.
+        For optional arguments check the underlying definition of
+        :meth:`QueryLearn.query_learn_header`.
+        :param query_type: Query type (number according to manual)
+        :type query_type: int or str
+        """
+        return QueryLearn.query_learn_header(
+            self.ask, query_type, self._smu_references, **kwargs)
+
+    def reset(self):
```
Comment (bilderbuchi) [inline pymeasure/instruments/hp/hp4142B.py:107]: nowadays that exists in `Instrument`, so should be removed in both classes.

File: pymeasure/instruments/hp/hp4142B.py [hunk 2]
```diff
@@ -0,0 +1,2012 @@
...
+        """
+        self.write("AB")
+
+    def force_gnd(self):
+        """ Force 0V on all channels immediately. Current Settings can
+        be restored with RZ. (``DZ``)
+        """
+        self.write("DZ")
+
+    def check_errors(self):
```
Comment (bilderbuchi) [inline pymeasure/instruments/hp/hp4142B.py:195]: nowadays that exists in `Instrument`, so should be removed in both classes.

File: pymeasure/instruments/hp/hp4142B.py [hunk 3]
```diff
@@ -0,0 +1,2012 @@
...
+        """
+        error = self.ask("ERR?")
+        error = error.split(",")
+        if int(error[0]) == 0:
+            return
+        else:
+            raise IOError(
+                "HP4142B Errors {0}, {1}, {2}, {3}".format(error[0], error[1], error[2], error[3]))
+
+    def check_idle(self):
```
Comment (bilderbuchi) [inline pymeasure/instruments/hp/hp4142B.py:206]: nowadays that exists as `Instrument.complete`, so should be removed in both classes.

**Comment by bmoneke** [comment]
@matthias6 do you want to pick up this PR or may I close it? It can be reopened whenever you want to continue.

**Comment by bmoneke** [comment]
Closed due to lack of author activity. It can be reopened if desired.

**Pull request closed without merging**